tests-scan

```diff
@@ -38,12 +38,14 @@ try:
 except ImportError:
     no_amqp = True
 
+Policy = Mapping[str, Sequence[str]]
 
-def build_policy(repo, context):
+
+def build_policy(repo: str, requested_contexts: Iterable[str]) -> Policy:
```

Comment:
The use of `Iterable` looks suspicious here because you can't always do a truth compare on `Iterable` to find out of its empty (which the code does, below).

Probably you want `Collection`, `Sequence` would also likely be fine, but it's overkill.

In general, I try to avoid `Iterable` or `Iterator` on interfaces, because it means that the value you receive might possibly be destroyed by your interaction with it.  Note specifically that an `Iterator` is `Iterable` (which is the problem).  `Collection` is the type which means "durable collection of things that I can iterate over".

This stuff is pretty opaque, so a quick primer on some confusingly-overlapping abstract types, from strict to loose:

 - `Sequence`: implements `in`, has a size, has a specific order, and can be indexed.  This is basically a `tuple` or a `list`.  Only use this if the order actual matters.
 - `Container`: implements `in`, has a size, but no specific order and cannot be indexed.  `tuple` and `list` are `Container`, but `set` is too (whereas `set` is not a `Sequence`).  This one tends to be the one you want most of the time when the exact order isn't important.
 - `Collection`: implements `in` but is not iterable.  Not super useful.

Reply:
Thanks for the primer! Collection and Container don't work here, as neither is iterable (and we use a `for` loop over it). As this isn't public API, I went with `Sequence` which is both fixed and iterable, the "overkill" doesn't matter here (it's basically a `list` and nothing else..)

Reply:
hilariously, I flipped `Collection` and `Container` in my own primer :facepalm: .  `Collection` is indeed iterable.

Reply:
Heh -- Collection it is then. Utterly unimportant here, but a good learning exercise.